lib/irb/pager.rb

```diff
@@ -7,18 +7,21 @@ class Pager
     PAGE_COMMANDS = [ENV['RI_PAGER'], ENV['PAGER'], 'less', 'more'].compact.uniq
 
     class << self
-      def page_content(content)
+      def page_content(content, retain_content: false)
         if content_exceeds_screen_height?(content)
-          page do |io|
+          page(retain_content: retain_content) do |io|
             io.puts content
           end
         else
           $stdout.puts content
         end
       end
 
-      def page
+      def page(retain_content: false)
         if IRB.conf[:USE_PAGER] && STDIN.tty? && pager = setup_pager
+          # if retain_content is true, we want to print the content to the designated IO too, which is $stdout for now
+          yield $stdout if retain_content
```

Comment:
It looks like `more` behaves similar to `less -X -F`
`echo hello | more` exits just after printing.
`ruby -e "puts 100.times.to_a" | more` and press `q`, output remains.

When user uses `more`, print and show pager will double-print the output.

Reply:
I see. I can't find `more` on macOS so I can't test it 😞 
But given this I decided to implement content retaining  with `-X` flag for now as we're already appending the `-R` flag. I didn't add `-F` flag though as I think the current pager already does the size check itself?

Reply:
It's weird that on my machine the `-R -X` does exactly what I expect. But on CI the same command doesn't preserve the output after quitting pager:

```
Failure: test_long_evaluation_output_is_preserved_after_paging(IRB::RenderingTest::TestVTerm):
  </(a{80}\n){6}/> was expected to be =~
  <"start IRB\n" +
  "irb(main):001> 'a' * 80 * 11\n" +
  "Paging content with command: [\"less\", \"-R\", \"-X\"]\n" +
  "WARNING: terminal is not fully functional\n" +
  "Press RETURN to continue\n" +
  "irb(main):002> 'foo' + 'bar'\n" +
  "=> \"foobar\"\n" +
  "irb(main):003>\n">.
```

Reply:
To get consistent result on both local and CI environment, I think we need to start setting `TERM` in rendering tests. And since we want to make these tests as close to most users' setup as possible, maybe it's better to set it to something like `xterm-256color` instead of `dumb`?